TypeScript not choosing the correct type for Instrument V5

When using the following code:

```typescript
// ... { category: 'linear' }
const [instrumentInfo] = response.result.list;
const stepSize = instrumentInfo.lotSizeFilter.qtyStep;
```

I get the following error message:

```
error TS2339: Property 'qtyStep' does not exist on type '{ maxOrderQty: string; minOrderQty: string; qtyStep: string; postOnlyMaxOrderQty?: string | undefined; } | { maxOrderQty: string; minOrderQty: string; qtyStep: string; } | { basePrecision: string; ... 4 more ...; maxOrderAmt: string; }'.
  Property 'qtyStep' does not exist on type '{ basePrecision: string; quotePrecision: string; minOrderQty: string; maxOrderQty: string; minOrderAmt: string; maxOrderAmt: string; }'.
```

Currently this is fixed by casting the object to `LinearInverseInstrumentInfoV5` like so:

```typescript
// ... { category: 'linear' }
const [instrumentInfo] = response.result.list;
const stepSize = (instrumentInfo as LinearInverseInstrumentInfoV5).lotSizeFilter.qtyStep;
```

When looking through the code, I see that the problem probably lies on `src/types/response/v5-market.ts` on lines `113-116`, see sample:

```typescript
export type InstrumentInfoResponseV5 =
  | CategoryCursorListV5<LinearInverseInstrumentInfoV5[], 'linear' | 'inverse'>
  | CategoryCursorListV5<OptionInstrumentInfoV5[], 'option'>
  | CategoryCursorListV5<SpotInstrumeI'mntInfoV5[], 'spot'>;
```

I've tried fixing this in some local commit but I can't seem to get it working, perhaps my knowledge with TypeScript doesn't reach this far yet. I also would not be surprised if this same issue persists for other methods with a shared return type.

Could someone take a look at this please? Thanks!

--- a/src/rest-client-v5.ts
+++ b/src/rest-client-v5.ts
@@ -231,9 +231,9 @@ export class RestClientV5 extends BaseRestClient {
    *
    * Note: Spot does not support pagination, so limit & cursor are invalid.
    */
-  getInstrumentsInfo(
-    params: GetInstrumentsInfoParamsV5,
-  ): Promise<APIResponseV3WithTime<InstrumentInfoResponseV5>> {
+  getInstrumentsInfo<C extends CategoryV5>(
+    params: GetInstrumentsInfoParamsV5 & { category: C },
+  ): Promise<APIResponseV3WithTime<InstrumentInfoResponseV5<C>>> {
     return this.get('/v5/market/instruments-info', params);
   }
 

--- a/src/types/response/v5-market.ts
+++ b/src/types/response/v5-market.ts
@@ -1,6 +1,8 @@
 import {
   CategoryCursorListV5,
+  CategoryV5,
   ContractTypeV5,
+  CursorListV5,
   InstrumentStatusV5,
   OptionTypeV5,
   OrderSideV5,
@@ -110,10 +112,14 @@ export interface SpotInstrumentInfoV5 {
   };
 }
 
-export type InstrumentInfoResponseV5 =
-  | CategoryCursorListV5<LinearInverseInstrumentInfoV5[], 'linear' | 'inverse'>
-  | CategoryCursorListV5<OptionInstrumentInfoV5[], 'option'>
-  | CategoryCursorListV5<SpotInstrumentInfoV5[], 'spot'>;
+type InstrumentInfoV5Mapping = {
+  linear: LinearInverseInstrumentInfoV5[],
+  inverse: LinearInverseInstrumentInfoV5[],
+  option: OptionInstrumentInfoV5[],
+  spot: SpotInstrumentInfoV5[]
+};
+
+export type InstrumentInfoResponseV5<C extends CategoryV5> = CategoryCursorListV5<InstrumentInfoV5Mapping[C], C>;
 
 export default interface OrderbookLevelV5 {
   price: string;

--- a/src/types/response/v5-market.ts
+++ b/src/types/response/v5-market.ts
@@ -2,7 +2,6 @@ import {
   CategoryCursorListV5,
   CategoryV5,
   ContractTypeV5,
-  CursorListV5,
   InstrumentStatusV5,
   OptionTypeV5,
   OrderSideV5,

